### PR TITLE
Add template build time stats

### PR DESCRIPTION
**Pull request by ammario**
https://user-images.githubusercontent.com/7416144/196004621-e1d8c03e-c398-4264-bac1-05d3c0fb4e99.mov

image

image

### CODE REVIEW

File: coderd/metricscache/metricscache.go [hunk 1]
```diff
@@ -138,9 +141,24 @@ func (c *Cache) refresh(ctx context.Context) error {
 		}
 		templateDAUs[template.ID] = convertDAUResponse(rows)
 		templateUniqueUsers[template.ID] = countUniqueUsers(rows)
+		avgBuildTime, err := c.database.GetTemplateAverageBuildTime(ctx, database.GetTemplateAverageBuildTimeParams{
```
Comment (kylecarbs) [inline coderd/metricscache/metricscache.go:144]: Other vars are verbose and prefixed with `template`, so we might wanna do that here.

File: site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import LinearProgress from "@material-ui/core/LinearProgress"
```
Comment (kylecarbs) [inline site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:1]: We should add stories here. I could easily see something accidentally happening and this not looking nice.

**Review by kylecarbs — APPROVED** [review]
Can you post a GIF of what this looks like?

File: site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx (hunk 2 above)
Comment (ammario) [reply site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:1]: It's covered in the Workspace Starting state story.

Comment (kylecarbs) [reply site/src/components/WorkspaceBuildProgress/WorkspaceBuildProgress.tsx:1]: Reasonable

**Pull request merged**